docs/cns.md

```diff
@@ -0,0 +1,110 @@
+# Chain Name Service Draft
+
+The goal of this module is to be a simple registry with the ability to transfer ownership. Auctions and other mechanisms are out of scope and can be implemented alongside or on top of this system (perhaps, in a custodial way).
+
+# State
+
+## Chain name
+
+Chain name contains information about the chain, most importantly, chain ID used to uniquly identify the chain and a list of peer nodes.
+
+Key: `chain-[chainID] -> Chain Name` 
+
+```go
+type ChainName struct {
+	ChainID string
+	Owner sdk.AccAddress
+	Peers []peer
+	SourceURL string
+	SourceCommitHash string
```

Comment:
what is this?

Reply:
I think hes saying genesis file location?

Reply:
Commit hash of the source code.

Reply:
This could also be the hash of the binary (but this is much the same as commit hash)

Reply:
I would also think it is necessary to add a genesis hash

Reply:
I could imagine a genesis hash is more valuable than a binary hash. There would ideally be multiple binaries for operating the node anyway, or some customization at a non-state-breaking level. The genesis hash needs to be consistent across them though.

Reply:
Agree having genesis hash is more useful, but I also feel it's better to include a commit hash as going forward chains using upgrade module will not have genesis files being changed but just the binaries